examples/usage_example.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	golog "log"
+	"log/slog"
+	"time"
+
+	"github.com/ONSdigital/log.go/v3/log"
+	pkgerrors "github.com/pkg/errors"
+	"golang.org/x/xerrors"
+)
+
+var exampleStruct = struct {
+	A string
+	B int
+}{"aaaa", 123}
+
+func main() {
+	log.Initialise("v3example")
+
+	ctx := context.Background()
+
+	slog.Info("slog with attrs", slog.String("a", "b"), slog.Int("int", 123))
+
+	slog.Info("slog with group", slog.Group("stuff", slog.String("a", "b"), slog.Int("int", 123)))
+
+	// Simple Info
+	log.Info(ctx, "v3 simple info")
+
+	// Complex Info
+	log.Info(ctx, "v3 info with data", log.Data{"key": "value", "struct": exampleStruct})
+
+	// Error
+	err := errors.New("some error")
+	log.Error(ctx, "v3 error", err)
+
+	// Error with extra data
+	log.Error(ctx, "v3 error with data", err, log.Data{"key": "value", "struct": exampleStruct})
+
+	// Fatal
+	log.Fatal(ctx, "v3 fatal", err)
```

Comment:
This indicates log.fatal would stop the app:
https://www.google.co.uk/search?q=golang+does+log.fatal+stop+program&sca_esv=5caaf0c1b0a489d0&sca_upv=1&source=hp&ei=d-6pZvu2Oqy5hbIPjNKYwAg&iflsig=AL9hbdgAAAAAZqn8h-0XFzI3Un-jqjhcpk-__uJPNupj&ved=0ahUKEwi7zveX59CHAxWsXEEAHQwpBogQ4dUDCA8&uact=5&oq=golang+does+log.fatal+stop+program&gs_lp=Egdnd3Mtd2l6IiJnb2xhbmcgZG9lcyBsb2cuZmF0YWwgc3RvcCBwcm9ncmFtMgUQIRigATIFECEYoAEyBRAhGKABSPZhUKsJWMpecAF4AJABAJgBtAGgAbUWqgEEMzEuM7gBA8gBAPgBAZgCI6ACoBeoAgrCAhAQLhgDGOUCGOoCGIwDGI8BwgIQEAAYAxjlAhjqAhiMAxiPAcICCxAAGIAEGLEDGIMBwgIREC4YgAQYsQMY0QMYgwEYxwHCAgUQABiABMICDhAAGIAEGLEDGIMBGIoFwgIIEAAYgAQYsQPCAgsQLhiABBjHARivAcICDhAuGIAEGLEDGIMBGIoFwgIIEC4YgAQYsQPCAg4QLhiABBixAxjRAxjHAcICCxAuGIAEGLEDGIMBwgIEEAAYA8ICBhAAGBYYHsICCxAAGIAEGIYDGIoFwgIIEAAYgAQYogTCAgQQIRgVwgIFECEYnwXCAgcQIRigARgKmAMIkgcEMzAuNaAHn6wB&sclient=gws-wiz

Does this log.Fatal() stop the application ?
If Yes then it needs moving to the end ?
If No, then why does it diverge form the norm ?

Reply:
The standard library `log.Fatal` does indeed stop the app. However, the log.go implementation never has (in V1 or V2) so I kept the same behaviour in V3 as it's supposed to be a near drop-in replacement. The [standard library log/slog package](https://pkg.go.dev/log/slog#pkg-functions) doesn't have a Fatal function at all. Note that the example usage above imports log.go as `log` but imports the standard library log as `golog`. This does cause some linting issues, but [we address that in the cilint config where needed](https://github.com/search?q=org%3AONSdigital+path%3A.golangci.yml+Fatal&type=code).

Reply:
Thanks for the explanation, shame our log.Fatal does not stick to the norm.